CMakeLists.txt

```diff
@@ -18,7 +18,19 @@ project(syslog-ng C)
 
 option(ENABLE_CPP "Enable C++" ON)
 
+if (ENABLE_CPP)
+  enable_language(CXX)
+  set(CMAKE_CXX_STANDARD 14 CACHE STRING "C++ standard")
```

Comment:
I do not know why is this different again compared to the autotools version, but can't we create a construct like this here as well?

```
if test "x$enable_cpp" = "xyes"; then
        AX_CXX_COMPILE_STDCXX(11,, mandatory)
        AX_CXX_COMPILE_STDCXX(17,, optional)
elif test "x$enable_cpp" = "xauto"; then
        AX_CXX_COMPILE_STDCXX(11,, optional)
        if test "x$HAVE_CXX11" = "x1"; then
                enable_cpp=yes
        else
                enable_cpp=no
        fi
        AX_CXX_COMPILE_STDCXX(17,, optional)
fi
```

Reply:
There is a way to specify minimum required and desired C++ standard in CMake.

P. S. I think I only added a macro there, not changed C++ standard defaults. Away from the machine atm, difficult to check from mobile.

Reply:
no offense, I know it was not you, just cannot see why there is a difference here

Reply:
as a first-round "fix" could you please just update

set(CMAKE_CXX_STANDARD 14 CACHE STRING "C++ standard")

to

set(CMAKE_CXX_STANDARD 11 CACHE STRING "C++ standard")


after that i will approve and merge the PR

thanks!

Reply:
Sure, will do.

(C++11 won’t work for some modules, possibly, since at least Abseil and friends use C++17 now.)

Reply:
never mind, i will do it in one run